### PR TITLE
Bing news fixes

### PR DESCRIPTION
**Pull request by raghavCompile**
Updating and fixing bugs in PyBingNewsSearch for adding data to ccagent.

### CODE REVIEW

File: py_bing_search/py_bing_search.py [hunk 1]
```diff
@@ -91,10 +91,10 @@ def search_all(self, query, format='json', limit=100, **kwargs):
         return results
 
     def _search(self, query, format='json', **kwargs):
-        kwargs['$format'] = format
-        url = self.QUERY_URL.format(urllib2.quote("'{}'".format(query)))
+        url = self.QUERY_URL.format(urllib2.quote("'{}'".format(query)), 'json')
         r = requests.get(url, auth=("", self.api_key), params=kwargs)
 
+        print r.url
```
Comment (gingerjoos) [inline py_bing_search/py_bing_search.py]: Why a print here?

Comment (raghavCompile) [inline py_bing_search/py_bing_search.py]: Added as a part of debug. Will remove it.

File: py_bing_search/py_bing_search.py [hunk 2]
```diff
@@ -115,19 +119,34 @@ def search_latest(self, query, format='json', **kwargs):
...
-        while crawling:
+        current_url = None
+        while True:
             kwargs['$skip'] = len(results)
             more_results = self._search(query, format=format, **kwargs)
             for result in more_results:
-                current_date = dateutil.parser.parse(result.date).date()
+                try:
+                    date = result.date
+                except:
```
Comment (gingerjoos) [inline py_bing_search/py_bing_search.py]: bare excepts are not allowed

File: py_bing_search/py_bing_search.py [hunk 3]
```diff
@@ -115,19 +119,34 @@ def search_latest(self, query, format='json', **kwargs):
...
+                current_date = dateutil.parser.parse(date).date()
                 if current_date < before_date:
-                    crawling = False
-                    break
-                results.append(result)
+                    results.append(result)
+            prev_url = current_url
+            try:
+                current_url = more_results[-1].url
+            except:
```
Comment (gingerjoos) [inline py_bing_search/py_bing_search.py]: same as above

File: py_bing_search/py_bing_search.py [hunk 4]
```diff
@@ -115,19 +119,34 @@ def search_latest(self, query, format='json', **kwargs):
...
+                current_url = more_results[-1]['Url']
+            if prev_url == current_url:
+                break
 
         return results
 
+    def get_query_url(self, query, **kwargs):
```
Comment (gingerjoos) [inline py_bing_search/py_bing_search.py]: where is this used?

Comment (raghavCompile) [inline py_bing_search/py_bing_search.py]: Used for updating query_url in OutboundAPILog

Comment (gingerjoos) [inline py_bing_search/py_bing_search.py]: Can't you add code in _search() method to handle this?

**Pull request merged**